### DISCUSSION TITLE
Verständnisfrage: Wie benutze ich Bindings um Datenpunkte in einer Statelist auszublenden?

### DISCUSSION THREAD

**Discussion by stony2k** [Zefau/ioBroker.jarvis#1263] [Q&A]
Ich habe mir jetzt die Postings hier und auch Beispiele im Wiki von mcuiobroker angeschaut, aber bekomme es nicht hin.

Ich würde den Datenpunkt "Sind Batterien leer" gerne ausblenden, sofern dieser gleich 0 ist:
![image](https://user-images.githubusercontent.com/16526471/144991300-ce9b173c-3040-4abe-99fd-cb33c9af2a25.png)

Versucht habe ich Gerät, bei der Eigenschaft Body Stil folgende Dinge ausprobiert:
{"{javascript.0.BatterieUeberwachung.EmptyBatCount}=0":{"display":"none !important"}}
{"{javascript.0.BatterieUeberwachung.EmptyBatCount}=0":{"visibility":"hidden"}}
{"0":{"visibility":"hidden"}}
{"0":{"display":"none !important"}}

Ich spreche hier ja keinen anderen Datenpunkt an, daher müsste "{javascript.0.BatterieUeberwachung.EmptyBatCount}" ja eig. nicht erforderlich sein.
![image](https://user-images.githubusercontent.com/16526471/144991687-aef8831b-d7aa-4902-ae87-e22565f15b9e.png)

Ich habe doch irgendwo einen totalen Denkfehler, oder?

**Comment by stony2k** [comment]
Neue Erkenntnis: Mit "{"0":{"display":"none !important"}}" wird bei 0 das Feld im Popup ausgeblendet. Da kann es von mir aus stehen bleiben, ich will es komplett aus der Statelist ausblenden.
![image](https://user-images.githubusercontent.com/16526471/144993379-364221bb-50d7-4d9e-8a4d-362e735b6ee1.png)

**Comment by mcuiobroker** [comment]
Welche Version hast du, hab es mit beta.11 getestet und es funktioniert als Binding.
Und mit "Wert"-Angabe funktioniert auch.
Hast du mal ein Upload gemacht: ```iobroker upload jarvis``` auf der console
Und einmal die Objektdaten zum Objekt: ` javascript.0.BatterieUeberwachung.EmptyBatCount`

**Comment by stony2k** [comment]
Ich hatte es mit der Beta 10 getestet und jetzt mit der Beta 11. iobroker upload jarvis gerade in der Konsole gemacht. javascript.0.BatterieUeberwachung.EmptyBatCount händisch in den Objekten von 0 auf 1 und wieder zurück. Leider wird der Datenpunkt im Statelist Widget nicht aktualisiert. Du kannst es dir gerne auch mal Live angucken, soll ich dich mal im Forum zwecks Kontakt anschreiben? Evtl. finden wir ja das Problem heraus und können es dann hier als Lösung posten.

**Comment by mcuiobroker** [comment]
Beispiel:
![image](https://user-images.githubusercontent.com/84818934/145049464-eacbde78-7d8b-4cd6-bf49-e207cdc9e14f.png)

**Comment by mcuiobroker** [comment]
![image](https://user-images.githubusercontent.com/84818934/145051216-20925dc1-8db2-4857-9957-a90d65f4face.png)

![jarvis-bindingmitbeta11](https://user-images.githubusercontent.com/84818934/145051098-6df4f296-db16-4e54-b378-41157d20a2f6.gif)
```
{
  "common": {
    "name": "0_userdata.0.testwert",
    "role": "state",
    "type": "number",
    "smartName": false
  },
  "native": {},
  "type": "state",
  "_id": "0_userdata.0.testwert",
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  },
  "from": "system.adapter.admin.0",
  "user": "system.user.admin",
  "ts": 1638888273830
}
```

**Reply by mcuiobroker** [reply]
Du kannst ja mal Deine Version darstellen mit ScreenToGif?

**Comment by stony2k** [comment]
Objektdaten:
```
{
  "common": {
    "read": true,
    "write": true,
    "name": "Zähler für Anzahl der zu wechselnden Batterien",
    "type": "number",
    "role": "state",
    "def": 0
  },
  "native": {},
  "type": "state",
  "_id": "javascript.0.BatterieUeberwachung.EmptyBatCount",
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  },
  "from": "system.adapter.admin.0",
  "user": "system.user.admin",
  "ts": 1638835492562
}
```

Eigenschaften Gerät Datenpunkt:
![image](https://user-images.githubusercontent.com/16526471/145074701-c5d601b7-dc06-4788-9bf2-3191690b949a.png)

GIF Animation:
![Animation](https://user-images.githubusercontent.com/16526471/145074747-f55870b4-4525-4e97-bb01-b5c47747d9a3.gif)

**Comment by mcuiobroker** [comment]
Anstatt 
![image](https://user-images.githubusercontent.com/84818934/145075830-aa4b24fc-c376-45bf-90cf-1a7fcb286729.png)
dort 
{"0":{"display":"none !important"}}
Falls das nicht funktioniert F12 ist etwas im Log zu sehen?
Unterschiede im Objekt sind ja nur
```
    "read": true,
    "write": true,
    "def": 0
```
Dann müsste @zefau sich das nochmal anschauen.

**Comment by stony2k** [comment]
Vielen Dank schonmal für deine Mühe :) Es ist doch echt zum verrückt werden :D Soll ich hieraus also am besten eine Issue generieren?

Mit {"0":{"display":"none !important"}} das gleiche Ergebnis leider. Es wird nur das Feld im Popup ausgeblendet, habe es nochmal mit aufgenommen.
![2](https://user-images.githubusercontent.com/16526471/145077097-129e78c2-6ab0-4637-869a-e5c223060784.gif)

Im Log sehe ich jetzt nichts, was auf das Problem hindeuten könnte, zumindest nicht wissentlich:
![image](https://user-images.githubusercontent.com/16526471/145077253-4a19d409-f7b9-4965-b0a6-a11264afd127.png)
![image](https://user-images.githubusercontent.com/16526471/145077309-3e40c9fb-ad04-47fb-9572-7d0fa6e7e037.png)

**Reply by mcuiobroker** [reply]
Danke, dass müsste erstmal an Infos zur Bearbeitung reichen. Sonst wird sich @Zefau sicherlich melden.

**Reply by mcuiobroker** [reply]
Einen Issue kannst du gerne mit Referenz öffnen.

**Reply by mcuiobroker** [reply]
Eins fällt mir noch ein. Kannst du dieses Widget:
![image](https://user-images.githubusercontent.com/84818934/145078779-93a02dd7-341d-4e37-bd0e-cd97e65d4e31.png)
als JSON aus dem Layout kopieren.

**Reply by stony2k** [reply]
Gerne, hoffe das ist so richtig rauskopiert. Die ID **fcb15739-1f8e-427a-81eb-42cc18c2d720** ist das betroffene Gerät.
```
  "7af1647a-fd2f-4b37-a128-7fd42c843715": {
    "id": "7af1647a-fd2f-4b37-a128-7fd42c843715",
    "config": {
      "dense": false
    },
    "icon": "window-closed-variant",
    "title": "Fenster & Türen",
    "module": "StateList",
    "revision": "21ca6",
    "items": [
      {
        "type": "device",
        "deviceId": "ef12cf22-0856-4995-91d2-859ae3be062b",
        "id": "10e8e53c-30ab-46b2-9f35-206e468a1a8d",
        "primaryStateKey": "",
        "bodyStateKey": "open",
        "bodyElement": "LastChangeBody",
        "secondaryStatesKeys": [],
        "LastChangeBodyConfig": {
          "open": {
            "time": "lc"
          }
        }
      },
      {
        "type": "device",
        "deviceId": "2a30fc84-014b-45b5-b69b-14334782e3a0",
        "id": "7c3245b5-8805-4345-ae13-c16d0b19ba13",
        "primaryStateKey": "",
        "bodyStateKey": "open",
        "bodyElement": "LastChangeBody"
      },
      {
        "type": "device",
        "deviceId": "0471658e-aa18-43ed-bde3-b8e5a7906b6b",
        "id": "117cebf0-8945-4356-8e70-4a624275b0e2",
        "primaryStateKey": "",
        "bodyStateKey": "open",
        "bodyElement": "LastChangeBody"
      },
      {
        "type": "device",
        "deviceId": "7cd7871f-2ec8-419a-a9aa-8357a49152d8",
        "id": "55552593-6a09-4d26-876f-8da435904666",
        "primaryStateKey": "",
        "bodyStateKey": "open",
        "bodyElement": "LastChangeBody"
      },
      {
        "type": "device",
        "deviceId": "40481b62-c457-4a56-8879-969002a67ef2",
        "id": "99384377-83c0-404d-84b4-d77a88aad05e",
        "primaryStateKey": "",
        "bodyStateKey": "open",
        "bodyElement": "LastChangeBody",
        "label": "Wohnungstür"
      },
      {
        "type": "device",
        "deviceId": "wohnungstür_9528d",
        "id": "0cae350c-1d7e-4414-86cb-9249c0791237",
        "primaryStateKey": "",
        "bodyStateKey": "lockState",
        "bodyElement": "LastChangeBody",
        "label": "Tüschloss",
        "secondaryStatesKeys": []
      },
      {
        "type": "device",
        "deviceId": "fcb15739-1f8e-427a-81eb-42cc18c2d720",
        "id": "f9d79219-7c10-40df-a211-226eb9fd8152",
        "primaryStateKey": "",
        "bodyStateKey": "batterycount",
        "bodyElement": "LastChangeBody",
        "label": ""
      }
    ],
    "hideTitle": true,
    "label": "Hauptseite: Fenster & Türen"
  }
```
